src/type.js

```diff
@@ -514,6 +514,9 @@ Sk.builtin.type["$r"] = function () {
         return new Sk.builtin.str("<type 'type'>");
     }
 };
+Sk.builtin.type.tp$setattr = function(obj, pyName, value) {
+    throw new Sk.builtin.TypeError("can't set attributes of built-in/extension type '" + obj.tp$name + "'");
```

Comment:
We're pretty confident we know what `obj.tp$name` is here, right? `obj` will be `Sk.builtin.type`!

However, this three-line function is useful in several places, hence the copy-and-paste. Perhaps it should be abstracted out? (Minor thing)

Reply:
I've changed this to `this.tp$name`
This is currently correct but would need to adjust depending on how #1079  is resolved...

Reply:
I could change this to `this.prototype.tp$name`